MetaMask/metamask-extension
# Fix switch-ethereum-chain handler by passing configuration id to setA…

Fixes https://github.com/MetaMask/metamask-extension/issues/18453

## Explanation

In a previous commit, we updated `setActiveNetwork` in the network controller to receive a network configuration id instead of a whole network object. However, the parameter passed to `setActiveNetwork` in the `switch-ethereum-chain.js` handler was still passed the object, not the id. See https://github.com/MetaMask/metamask-extension/commit/23ca4460cf#diff-93f17dab5026baf840aec766ac5b6ebcd35d5bb0abb45e2c3f41791423f2a488R117

This PR corrects that, and adds an e2e test for the `wallet_switchEthereumChain` functionality, which would have caught this bug.

It is worth noting that the bug this PR fixes would not occur if you switched to the network immediately after adding it, because the correct parameter is passed to the `setActiveNetwork` call in `add-ethereum-chain.js` which happens after prompting the user to switch networks (https://github.com/MetaMask/metamask-extension/commit/23ca4460cf#diff-b3485a1ea9943fd380364f73a53295289f2ccd3817423d4f245b5102f8620b3cR170)

## Manual Testing Steps

1. Run `$(yarn bin ganache) --port 8546 --chain.chainId 1338`
2. Go to https://metamask.github.io/test-dapp/ and connect to the dapp
3. Click "Add Localhost 8546" and "Approve", BUT click "Cancel" once you get to the "Allow this site to switch the network?" screen (so that if you were to open MetaMask, the Localhost 8546 network would not be selected)
4. Click "Switch to Localhost 8546" and then "Approve"
5. If you open MetaMask, the Localhost 8546 network should now be selected

## Pre-merge author checklist


## Pre-merge reviewer checklist


If further QA is required (e.g. new feature, complex testing steps, large refactor), add the `Extension QA Board` label.

In this case, a QA Engineer approval will be be required.
